File: Practica 2/Practica2.java
import javax.swing.*;
import java.awt.*;

public class Practica2 {

    // Clase Punto
    public static class Punto {
        public float x;
        public float y;

        public Punto(float x, float y) {
            this.x = x;
            this.y = y;
        }

        public String coord_cartesianas() {
            return "(" + x + ", " + y + ")";
        }

        public String coord_polares() {
            double r = Math.sqrt(x * x + y * y);
            double theta = Math.atan2(y, x);
            return ("Radio: " + r + " Angulo: " + theta);
        }

        public String toString() {
            return "Punto: " + x + ", " + y;
        }
    }

    // Clase Linea
    public static class Linea extends JPanel {
        public Punto p1;
        public Punto p2;

        public Linea(Punto p1, Punto p2) {
            this.p1 = p1;
            this.p2 = p2;
        }

        public String toString() {
            return "Linea desde " + p1 + " hasta " + p2;
        }

        public void paintComponent(Graphics g) {
            super.paintComponent(g);
            Graphics2D g2d = (Graphics2D) g;
            g2d.drawLine((int) p1.x + 50, (int) p1.y + 50, (int) p2.x + 50, (int) p2.y + 50);

        }

        public static void dibujaLinea(Punto p1, Punto p2) {
            JFrame frame = new JFrame("Dibujando una Línea");
            Linea panel = new Linea(p1, p2);
            frame.add(panel);
            frame.setSize(400, 400);
            frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
            frame.setVisible(true);
        }
    }

    // Clase Circulo
    public static class Circulo extends JPanel {
        public Punto centro;
        public float radio;

        public Circulo(Punto c, float r) {
            this.centro = c;
            this.radio = r;
        }

        public String toString() {
            return "Circulo con centro: " + centro + " y radio de: " + radio;
        }

        protected void paintComponent(Graphics g) {
            super.paintComponent(g);
            Graphics2D g2d = (Graphics2D) g;
            int x = (int) (centro.x - radio);
            int y = (int) (centro.y - radio);
            int diametro = (int) (2 * radio);
            g2d.drawOval(x, y, diametro, diametro);
        }

        public static void dibujaCirculo(Punto centro, float radio) {
            JFrame frame = new JFrame("Dibujando un Círculo");
            Circulo panel = new Circulo(centro, radio);
            frame.add(panel);
            frame.setSize(2 * (int) radio + 100, 2 * (int) radio + 100);
            frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
            frame.setVisible(true);
        }
    }

    public static void main(String[] args) {
        // Crear puntos para los objetos
        Punto p1 = new Punto(0, 0);
        Punto p2 = new Punto(5, 5);
        Punto centro = new Punto(200, 200);

        // Crear y mostrar la linea
        Linea linea = new Linea(p1, p2);
        System.out.println(linea.toString());
        Linea.dibujaLinea(p1, p2);

        // Crear y mostrar el círculo
        Circulo circulo = new Circulo(centro, 150.0f);
        System.out.println(circulo.toString());
        Circulo.dibujaCirculo(centro, 150.0f);
    }
}

File: Practica 2/Practica 2.py
import math
import matplotlib.pyplot as plt

class Punto:
    def __init__(self, x: float, y: float):
        self.x = x
        self.y = y

    def coord_cartesianas(self):
        return (self.x, self.y)
        
    def coord_polares(self):
        r = math.sqrt(self.x ** 2 + self.y ** 2)
        theta = math.atan2(self.y, self.x)
        return (r, theta)
    
    def __str__(self):
        return f"Punto {self.x}, {self.y}"
    

class Linea:
    def __init__(self, p1: Punto, p2: Punto):
        self.p1 = p1
        self.p2 = p2

    def __str__(self):
        return f"Linea desde {self.p1} hasta {self.p2}"
    
    def dibujeLinea(self):
        plt.plot([self.p1, self.p2], [self.p1, self.p2], marker='o', linestyle='-')
        plt.show()

class Circulo:
    def __init__(self, c: Punto, r: float):
        self.centro = c
        self.radio = r

    def __str__(self):
        return f"Circulo con centro {self.centro} y radio de {self.radio}"
    
    def dibujaCirculo(self):
        centro = (self.centro.x, self.centro.y)
        radio = self.radio
        
        fig, ax = plt.subplots()
        
        circulo = plt.Circle(centro, radio, color='b', fill=False)
        
        ax.add_patch(circulo)

        ax.scatter(self.centro.x, self.centro.y, color='r', s=10, label="Centro")

        ax.set_xlim(centro[0] - radio - 1, centro[0] + radio + 1)
        ax.set_ylim(centro[1] - radio - 1, centro[1] + radio + 1)

        ax.set_aspect('equal')

        plt.show()

#Salida Punto
p1 = (0,0)
p2 = (5,5)
linea = Linea(p1,p2)
print(linea)
linea.dibujeLinea()

#Salida Circulo
p = Punto(5,5)
circulo = Circulo(p, 3)
print(circulo)
circulo.dibujaCirculo()   